src/plugins/discover/server/ui_settings.ts

```diff
@@ -68,6 +70,19 @@ export const getUiSettings: (docLinks: DocLinksServiceSetup) => Record<string, U
     category: ['discover'],
     schema: schema.number(),
   },
+  [SAMPLE_ROWS_PER_PAGE_SETTING]: {
+    name: i18n.translate('discover.advancedSettings.sampleRowsPerPageTitle', {
+      defaultMessage: 'Number of rows per page',
+    }),
+    value: DEFAULT_ROWS_PER_PAGE,
+    options: ROWS_PER_PAGE_OPTIONS,
+    type: 'select',
+    description: i18n.translate('discover.advancedSettings.sampleRowsPerPageText', {
+      defaultMessage: 'The number of rows per page to show in the table',
```

Comment:
I would suggest to replace `show in the table` to `show in the Document Explorer`. WDYT @jughosta ?

Reply:
I think we need to have a better distinction between "Number of rows per page" and "Number of rows".  

Is "Number of rows" the maximum number of rows to include in the document table?

Reply:
@gchaps Yes. A total number of rows which will be rendered in the grid. Hits count can be greater though.

Reply:
In other copy, we've removed the term "Document Explorer".  So how about:

**Number of rows per page**
The number of rows to display per page of the document table.

Reply:
Updated! Thanks @gchaps !

Reply:
Text LGTM.

Reply:
Good to know! I wasn't aware of that, but I think in that case it's a good idea to keep 'sample' so they stay together.

Reply:
I would do this:

**Number of rows per page**
The number of rows to display per page of the document table.

**Maximum number of rows**
The maximum number of rows to display in the document table.

Reply:
Here is another suggestion with less text.  

**Rows per page**
Limits the number of rows per page in the document table.

**Maximum rows per table**
Sets the maximum number of rows for the entire document table.

Reply:
@gchaps I like the second version better too, thanks! Updated:

<img width="956" alt="Screenshot 2022-07-18 at 11 03 28" src="https://user-images.githubusercontent.com/1415710/179480591-112aa45b-5fb5-4cbe-a649-1371feb7f537.png">